core/src/test/java/org/elasticsearch/common/xcontent/ObjectSpoolerTests.java

```diff
@@ -0,0 +1,271 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.common.xcontent;
+
+import org.elasticsearch.common.ParseField;
+import org.elasticsearch.common.ParsingException;
+import org.elasticsearch.common.bytes.BytesReference;
+import org.elasticsearch.test.ESTestCase;
+
+import java.io.IOException;
+
+import static java.util.Collections.emptySet;
+import static org.elasticsearch.common.xcontent.ObjectSpooler.constructorArg;
+import static org.hamcrest.Matchers.instanceOf;
+
+public class ObjectSpoolerTests extends ESTestCase {
+    /**
+     * Builds the object in random order and parses it.
+     */
+    public void testRandomOrder() throws Exception {
+        int times = 100;
+        for (int i = 0; i < times; i++) {
```

Comment:
I think we get enough coverage from the CI?

The larger problem that I have with the poor man's repeat is that there are subtle issues with non-reproducible iterations. An example of this can be seen in #16208. If a non-deterministic number of random draws are taken on the `n`th iteration, then all further iterations are not reproducible.